pkg/metrics/interfaces.go

```diff
@@ -0,0 +1,8 @@
+package metrics
+
+import "time"
+
+type MetricsTransformer interface {
+	GetOutlierIntervalsAndInterpolate(
```

Comment:
Asssuming you are creating a generic MetricTransformer interface, the fn name should be generic eg. Transform. The actual implmentation of 'GetOutlierIntervalsAndInterpolate' can be in the transform fn of a OutlierInterpolatorTransformer implmentation. That will make the transformers more generic.